code/modules/buildmode/submodes/save_area.dm

```diff
@@ -0,0 +1,51 @@
+/datum/mapGenerator/save_area
+	buildmode_name = "Save Area"
+	modules = list(/datum/mapGeneratorModule/save_area)
+	var/min_x = 0
+	var/min_y = 0
+	var/max_x = 0
+	var/max_y = 0
+
+/datum/mapGenerator/save_area/defineRegion(turf/Start, turf/End, replace = 0)
+	min_x = min(Start.x,End.x)
+	min_y = min(Start.y,End.y)
+	max_x = max(Start.x,End.x)
+	max_y = max(Start.y,End.y)
+	..()
+
+/datum/mapGeneratorModule/save_area
+	var/areaName = "default.dm"
+
+//This could be optimised by making turfs that are the same go in the same, but this is a quick bodge solution so yea, fun job for coder here :)
+/datum/mapGeneratorModule/save_area/generate()
+	var/datum/mapGenerator/save_area/L = mother
+	if(!istype(L))
+		return
+	//If someone somehow gets build mode, stop them from using this.
+	if(!check_rights(R_ADMIN))
+		message_admins("[ckey(usr)] tried to run the map save generator but was rejected due to insufficient perms.")
+		to_chat(usr, "<span class='warning'>You must have R_ADMIN privellages to use this.</span>")
+		return
+	//Emergency check
+	if(L.map.len > 1600)
+		var/confirm = alert("Uhm, are you sure, the area is quiet large?", "Run generator", "Yes", "No")
+		if(confirm != "Yes")
+			return
+	//Log just in case something happens
+	log_game("[ckey(usr)] ran the save level map generator on [L.map.len] turfs.")
+
+	//Step 1: Get the data (This can take a while)
+	var/dat = convert_map_to_tgm(L.map)
+
```

Comment:
Somewhere around here you should send a confirmation message to the person that it's doing stuff.